### ISSUE TITLE
roadmap: evolve Jev email triage into personalized inbox automation

### ISSUE THREAD

**Issue by sygtt** [sygtt/agentic-inbox#28]
## 目的

Issue #27 で導入する Jev-based email triage を、いきなり完全自動化せず、安全に評価・段階導入していくためのロードマップ。

実装単位は原則として各フェーズごとに別 Issue に切る。この Issue 自体は roadmap / design record として使い、個別実装の acceptance criteria は各子 Issue に持たせる。

起点: #27

---

## 基本原則

- Jev は意味特徴量の抽出器として扱う
- 最終アクションは deterministic policy / supervised model で決める
- 破壊的・外部的アクションは段階的に解禁する
- まず prediction と human action を蓄積し、精度を観測する
- accuracy ではなく誤判定コストを重視する
- 特に auto-file は false positive を極端に低くする
- feature schema / policy / model は versioning する
- manual override を常に優先する
- upstream との乖離を増やしすぎない

---

# Phase v0 — Structured triage foundation

Tracking: #27

### 目的

現在の GLM auto-summary を Jev structured triage に置き換え、学習・評価可能なデータを作る。

~~~text
Inbound email
  -> Jev features
  -> deterministic policy v1
  -> email_triage_analysis
  -> disposition:* tag
~~~

### ポイント

- folder は動かさない
- 通知しない
- Draftを作らない
- Agent chat に summary を残さない
- manual disposition を上書きしない
- prediction / feature / model version / policy version を保存

### Exit criteria

- 本番受信で安定して analysis が蓄積する
- triage failure がメール配送・保存を壊さない
- #27 acceptance criteria を満たす

---

# Phase v0.1 — Shadow evaluation & observability

### 目的

Jev feature schema と policy v1 が、実際のメール運用に合っているかを測る。

### 追加したいもの

- prediction と人間の後続操作を比較できる評価基盤
- predicted disposition / manual override / folder move / read / star 等の既存操作を観測
- feature version / policy version ごとの集計
- representative error cases を確認できる仕組み

ここではまだ自動 Archive しない。

### 評価指標

- auto-file precision
- action-required recall
- manual override rate
- Jev failure rate
- disposition 別 confusion matrix

最も高コストな誤りは「重要メールを auto-file と誤判定すること」とする。

### Exit criteria

- 十分な実メールサンプルが蓄積
- 主要 false positive / false negative パターンを把握
- feature schema v2 が必要か判断可能
- auto-file の安全な threshold 候補が得られる

---

# Phase v0.2 — Conservative auto-file

### 目的

高confidence・低リスク領域だけ、実際の Archive を自動化する。

### 初期対象

~~~text
bulk_marketing very high
AND requires_action very low
AND requires_reply very low
AND financial_impact very low
AND security_relevance very low
~~~

### 方針

- disposition:auto-file と folder move は分離したままにする
- auto-file executor は明示的な allow condition を持つ
- low confidence / ambiguous は Inbox 維持
- manual hold / action-required 等は自動 move より優先
- rollback 可能な Archive のみに限定し、Trash は自動化しない

### Exit criteria

- auto-file precision が十分高い
- 誤Archiveが実運用上許容できる水準
- rollback / audit が可能

---

# Phase v0.3 — Action-required notification

### 目的

disposition:action-required をタグから実用的な attention mechanism にする。

### 候補

- app 内 notification / badge
- high-priority view
- optional push notification
- deadline / urgency の表示

通知判定は disposition だけでなく urgency / has_deadline / financial_impact / security_relevance / direct_personal などを利用する。

notification precision と dismissal rate を測り、通知疲れを避ける。

---

# Phase v0.4 — Calendar / Todo suggestions

### 目的

Jev feature を downstream action proposal に活用する。

### Calendar

calendar_candidate が高いメールについて、日付・時刻・場所・タイトルを別の structured extraction で抽出し、登録候補を出す。

### Todo

requires_action が高いメールについて、action / deadline / source email を構造化して ToDo候補を出す。

v0.4 では原則 suggest / approve モデルとし、モデル判断だけで外部カレンダーやToDoへ自動作成しない。

---

# Phase v0.5 — Explicit feedback dataset

### 目的

普通に使っているだけで得られる implicit signal に加えて、学習しやすい label を蓄積する。

### 保存候補

- prediction
- human final disposition
- folder action
- notification accepted/dismissed
- calendar suggestion accepted/rejected
- todo suggestion accepted/rejected
- timestamp
- feature schema version
- policy version

追加テーブル候補は email_triage_feedback。ただし既存操作ログで十分なら新規テーブルを増やす前に再検討する。

### Exit criteria

- supervised learning 用 dataset を再現可能に export できる
- label leakage / ambiguous labels の扱いを定義

---

# Phase v1 — Personalized policy model

### 目的

手書き閾値から、実際の利用履歴に基づく個人最適化へ移行する。

### 初手

Logistic Regression を第一候補とする。

理由:

- 小さいデータでも扱いやすい
- feature weight を解釈できる
- inference が軽い
- debugging しやすい
- baseline として優秀

入力は Jev semantic features + deterministic metadata（sender/domain, envelope recipient, attachment presence, thread metadata 等）。

hard safety rules は ML より上位に残し、confidence / abstention zone を設ける。offline evaluation の後に shadow deploy し、threshold は action ごとの誤判定コストで決める。

データ量が増えたら Logistic Regression と LightGBM / XGBoost を比較する。

---

# Phase v1.1 — Active learning / uncertainty review

### 目的

モデルが迷うメールだけ人間確認へ回して、少ない操作で dataset の質を上げる。

例:

- P(auto-file) が境界付近
- low Jev confidence
- policy/model disagreement
- new sender/domain

これらを review queue に入れ、自動化率を上げつつ安全性を維持する。

---

# Phase v2 — Replace / distill Jev where justified

### 目的

十分な教師データが得られた後、固定タスクについて Jev より軽い専用モデルが有利か検証する。

### 比較候補

- Jev
- BERT / ModernBERT / DeBERTa 系 encoder classifier
- embedding + classical classifier
- small local model
- current personalized policy

### 判断軸

- precision / recall
- latency
- inference cost
- privacy
- operational complexity
- retraining cost
- adaptability to new categories

Jev を置き換えること自体を目的にしない。

~~~text
汎用性が必要 -> Jev
固定タスク + 十分な教師データ -> encoder classifier が有利か検証
~~~

---

# 将来の理想形

~~~text
                    deterministic metadata
                           |
Inbound email -> Jev ------+
                   |       |
                   v       v
              semantic feature vector
                           |
                           v
                 safety / hard rules
                           |
                           v
                personalized ML policy
                           |
             +-------------+-------------+
             |             |             |
          auto-file    attention       review
                         |
                    +----+----+
                    |         |
                 Calendar    Todo
                 proposal  proposal
~~~

各層の責務:

- Jev: 意味を測る
- deterministic code: 確実に分かる事実を測る
- ML: 個人の判断傾向を学ぶ
- hard rules: 安全性を守る
- executor: 実際の操作を行う

---

## Issue運用

各Phaseを実装するときは、この roadmap から個別 Issue を作り、この Issue にリンクする。

Phase の順序は固定ではない。v0.1 の評価結果によって feature schema や priority は変更してよい。

ロードマップの目的は予定を守ることではなく、なぜ今の設計が次の段階に繋がるのかを残すこと。
